src/views/home/components/amount-input/amount-input.view.tsx

```diff
@@ -0,0 +1,85 @@
+import { BigNumber } from "ethers";
+import { parseUnits } from "ethers/lib/utils";
+import { ChangeEvent, FC, useState } from "react";
+
+import useAmountInputtStyles from "src/views/home/components/amount-input/amount-input.styles";
+import Typography from "src/views/shared/typography/typography.view";
+import { Token } from "src/domain";
+
+interface onChangeParams {
+  amount: BigNumber;
+  isInvalid: boolean;
+}
+
+interface AmountInputProps {
+  token: Token;
+  balance: BigNumber;
+  fee: BigNumber;
+  onChange: (params: onChangeParams) => void;
+}
+
+const AmountInput: FC<AmountInputProps> = ({ token, balance, fee, onChange }) => {
+  const [value, setValue] = useState("");
+  const classes = useAmountInputtStyles(value.length);
+  const actualFee = token.symbol.includes("ETH") ? fee : BigNumber.from(0);
+
+  const getFixedTokenAmount = (amount: string, decimals = 18): string => {
+    const amountWithDecimals = Number(amount) / Math.pow(10, decimals);
+    return Number(amountWithDecimals.toFixed(decimals)).toString();
+  };
+
+  const updateAmountInput = (amount: BigNumber) => {
+    const newAmountWithFee = amount.add(actualFee);
+    const isNewAmountWithFeeMoreThanFunds = newAmountWithFee.gt(BigNumber.from(balance));
+    const areFundsExceededDueToFee =
+      isNewAmountWithFeeMoreThanFunds && amount.lte(BigNumber.from(balance));
+    const isAmountInvalid =
+      isNewAmountWithFeeMoreThanFunds || areFundsExceededDueToFee || amount.isZero();
```

Comment:
Why is `areFundsExceededDueToFee` in this condition?

Reply:
This one was a integration to explicit one error in wallet-ui but we can remove it here

Reply:
Right. `areFundsExceededDueToFee` makes sense only if later we are specific about the error being produced because of the fees. We can either use it for better error messages or remove it. Do as you prefer.